monai/apps/auto3dseg/transforms.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Hashable, Mapping
+
+import numpy as np
+import torch
+
+from monai.config import KeysCollection
+from monai.networks.utils import pytorch_after
+from monai.transforms import MapTransform
+
+
+class CheckLabelShaped(MapTransform):
+    """
+    Checks if segmentation label images (in keys) have the same spatial shape as the main image (in source_key),
+    and raise an error if the shapes are significantly different.
+    if the shapes are only slightly different (within an allowed_shape_difference in each dim), then quietly resize the label.
```

Comment:
May I ask why it should be done quietly?

Reply:
maybe the transform's name should be EnsureSameShape or EnsureShape...

Reply:
> May I ask why it should be done quietly?

it's designed to correct a tiny shape mismatch. We could show a message or warning, but this can  potentially print many statements for the user.. if you think we need it , let me know